cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -1014,13 +1014,24 @@ func (s *Server) fetchChartWithRegistrySecrets(ctx context.Context, chartDetails
 	chartID := fmt.Sprintf("%s/%s", appRepo.Name, chartDetails.ChartName)
 	log.Infof("fetching chart %q with user-agent %q", chartID, userAgentString)
 
+	// Look up the cachedChart cached in our DB to populate the tarball URL
+	cachedChart, err := s.manager.GetChartVersion(chartDetails.AppRepositoryResourceNamespace, chartID, chartDetails.Version)
+	if err != nil {
+		return nil, nil, status.Errorf(codes.Internal, "Unable to fetch the chart %s (version %s) from the namespace %q: %v", chartID, chartDetails.Version, chartDetails.AppRepositoryResourceNamespace, err)
+	}
+	var tarballURL string
+	if cachedChart.ChartVersions != nil && len(cachedChart.ChartVersions) > 0 && cachedChart.ChartVersions[0].URLs != nil {
+		tarballURL = cachedChart.ChartVersions[0].URLs[0]
```

Comment:
Similarly here. I just checked the bitnami index.yaml and see that the first URL is always the tarball, but worth a comment since it's not obvious when looking at this why the first URL can just be used. Perhaps just with a link to https://helm.sh/docs/topics/chart_repository/#the-index-file